src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -89,9 +89,27 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
 
         protected abstract Task<AuthenticateResult> HandleRemoteAuthenticateAsync();
 
-        protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         {
-            return Task.FromResult(AuthenticateResult.Fail("Remote authentication does not support authenticate"));
+            var authenticateContext = new AuthenticateContext(Options.SignInScheme);
+            await PriorHandler.AuthenticateAsync(authenticateContext);
+            if (authenticateContext.Accepted)
+            {
+                if (authenticateContext.Error != null)
+                {
+                    return AuthenticateResult.Fail(authenticateContext.Error);
+                }
+                // Compare the ClaimsIssuer to make sure the identity was produced by this provider and not by some other one.
+                if (string.Equals(authenticateContext.Principal?.Identity?.AuthenticationType, Options.ClaimsIssuer, StringComparison.Ordinal))
```

Comment:
I guess my point was more that the remote handler shouldn't care what is in the sign in identity. And if something like claims transformation changes the issuer would now block this behavior.